dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
I very much *do not* want DART in the business of defining a canonical set of material properties. I am cautiously optimistic about @PyryM's suggestion:

> I think materials should just be key/value stores that map string keys to [vec4 | texture | string] values. 

It is straightforward to allow the user to store arbitrary vertex properties. We could simply allow the user to store a `std::vector<T>` with `size()` equal to the number of vertices. I am less sure about providing an API to store textures. Hopefully @PyryM can clarify a few points:

- Assimp supports spherical, cylindrical, and uv mapping. Are all three used in practice?
- I also found some references to uvw-mapping. How does this differ from uv mapping?
- Are uv-coordinates stored as separate vertex properties or attached to the image asset?
- How are uv-coordinates associated with the corresponding assert?
- Are all textures 24-bit RGBA? Is it common to use a different bit depth or number of channels?

My only other concern is that `VisualAddon` is attached to `ShapeFrame`, not `Shape`. These properties only apply to `MeshShape` and are intimately related to the mesh, e.g. arrays of vertex properties must have the same length as the number of vertices in the mesh.

Alternatively, we could take the simpler route I suggested in #453:

> `MeshNode` should contain the bare minimum amount of information necessary to perform collision checking and a dynamical simulation. Additionally, DART should provide an `Uri` to the original resource that can be loaded for visualization. The user is responsible for loading any additional information, e.g. textures and material properties, that is necessary only for visualization.

This has two downsides. First, every mesh is loaded from disk twice: once for collision detection and once for visualization. Second, we cannot programmatically construct a textured or vertex-colored mesh. This latter limitation bothers me more than the former - @mklingen and I independently ran into this in our research when rendering a color map.